source/includes/connect/csot.rb

```diff
@@ -0,0 +1,31 @@
+# start-csot-overrides
+require 'mongo'
+
+# Replace the placeholder with your connection string
+uri = "<connection string>"
+
+# Sets a client-level timeout configuration
+options = { timeoutMS: 30000 }
+client = Mongo::Client.new(uri, options)
+
+begin
+  db = client.use('test-db')
+  collection = db[:test-collection]
+
+  # Perform a query with an operation-level timeout configuration,
+  # overriding the client-level configuration
+  docs = collection.find({}, timeout_ms: 10000).to_a
+
+  docs.each { |doc| puts doc }
+ensure
+  client.close
+end
```

Comment:
I strongly recommend using a block for this pattern; it will automatically ensure the client is closed. (On the other hand, if you're wanting to emphasize--for documentation purposes--the creation and disposal of a client, you can ignore my recommendation.)

```suggestion
Mongo::Client.new(uri, options) do |client|
  db = client.use('test-db')
  collection = db[:test-collection]

  # Perform a query with an operation-level timeout configuration,
  # overriding the client-level configuration
  docs = collection.find({}, timeout_ms: 10000).to_a

  docs.each { |doc| puts doc }
end
```

Reply:
Looks good, thanks!